libnavigation-core/src/main/java/com/mapbox/navigation/core/location/ReplayRouteLocationEngine.kt

```diff
@@ -30,7 +30,7 @@ class ReplayRouteLocationEngine(
     private lateinit var mockedLocations: MutableList<Location>
     private lateinit var dispatcher: ReplayLocationDispatcher
     private lateinit var replayLocationListener: ReplayRouteLocationListener
-    private lateinit var lastLocation: Location
+    private var lastLocation = Location(REPLAY_ROUTE)
```

Comment:
`lastLocation = Location(REPLAY_ROUTE)` will create a location with valid location coordinates (0,0)

this means when `beginReplayWith` starts, it will start in the ocean
```
    private fun beginReplayWith(callback: LocationEngineCallback<LocationEngineResult>) {
        route?.let {
            start(it, callback)
        } ?: point?.let {
            startRoute(it, lastLocation, callback)
        } ?: callback.onFailure(Exception("No route found to replay."))
    }
```

we should suspend until the first valid location is received to begin the replay

Reply:
it's a tricky point just to avoid crash, need set route before start routing